src/bindings/c/src/ov_core.cpp

```diff
@@ -174,20 +110,29 @@ ov_status_e ov_core_read_model_from_memory(const ov_core_t* core,
 ov_status_e ov_core_compile_model(const ov_core_t* core,
                                   const ov_model_t* model,
                                   const char* device_name,
-                                  const ov_properties_t* property,
-                                  ov_compiled_model_t** compiled_model) {
+                                  const size_t property_size,
+                                  ov_compiled_model_t** compiled_model,
+                                  ...) {
     if (!core || !model || !compiled_model) {
         return ov_status_e::INVALID_C_PARAM;
     }
 
     try {
+        ov::AnyMap property = {};
+        va_list args_ptr;
+        va_start(args_ptr, compiled_model);
+        for (size_t i = 0; i < property_size; i++) {
```

Comment:
THe same comment here, we need to understand that real size of `va_list == 2 * property_size`

Reply:
We don't have any good method to runtime check/get the actual argument number, so will be better to pass arguments number rather than property number?